src/HealthGPS/energy_balance_model.cpp

```diff
@@ -35,66 +46,163 @@ void EnergyBalanceModel::generate_risk_factors([[maybe_unused]] RuntimeContext &
 }
 
 void EnergyBalanceModel::update_risk_factors(RuntimeContext &context) {
-    auto age_key = core::Identifier{"age"};
-    for (auto &entity : context.population()) {
-        // Ignore if inactive, newborn risk factors must be generated, not updated!
-        if (!entity.is_active() || entity.age == 0) {
+    hgps::Population &population = context.population();
+    const size_t population_size = population.current_active_size();
+    double mean_sim_body_weight = 0.0;
+    double mean_adjustment_coefficient = 0.0;
+
+    // TODO: Compute target body weight.
+    const float target_BW = 100.0;
+
+    // Trial run.
+    for (auto &person : population) {
+        // Ignore if inactive.
+        if (!person.is_active()) {
```

Comment:
This isn't a big deal, but if you *really* care about performance, iterating over a big `std::vector` and only doing something if a flag is (un)set is not the most efficient way to do it, because your CPU is constantly branching. It's faster if you have two separate vectors (in this case, one for active and one for inactive) and you just iterate over the one you're interested in.

As I said, def not important for now, but this might be something to look at if we're trying to optimise in future. (The fact that each person is a separate object is also a red flag from a performance standpoint!)

Reply:
Right. AoS vs SoA. I haven't even touched the `Population` code there, but I suspect there's tons of optimisation in there.

Reply:
Good point @dalonsoa. I'm afraid I've already merged #158 but I'll do it for #160!

Reply:
After thinking about this some more, I'm wondering if these "person IDs" actually need to be stored *anywhere*. If you're simulating a population of e.g. a million people, it's really odd to represent that with a vector containing the numbers one to a million. Probably we can get away with just keeping track of the various "counts" (i.e., number of active people, dead people and emigrants).

A job for another day, though.